repos/emacs/update

```diff
@@ -8,19 +8,21 @@ cd $SCRIPTPATH
 function update_repo() {
     owner=$1
     repo=$2
+    branch=$3
     echo $repo
 
     # Get relevant data (commit id and timestamp) for the latest commit
-    commit_data=$(curl "https://github.com/$owner/$repo/commits/master.atom" | xmlstarlet sel -N atom="http://www.w3.org/2005/Atom" -t -m /atom:feed/atom:entry -v "concat(atom:id,'/',atom:updated)" -n | head -n 1)
+    commit_data=$(curl "https://github.com/$owner/$repo/commits/$branch.atom" | xmlstarlet sel -N atom="http://www.w3.org/2005/Atom" -t -m /atom:feed/atom:entry -v "concat(atom:id,'/',atom:updated)" -n | head -n 1)
 
     # Extract commit sha and build a version number based on date: YYYYMMDD.0
     commit_sha=$(echo $commit_data | cut -d '/' -f 2)
     version_number=$(echo $commit_data | cut -d '/' -f 3 | cut -d 'T' -f 1 | sed 's/-//g').0
 
     digest=$(nix-prefetch-url --unpack "https://github.com/$owner/$repo/archive/${commit_sha}.tar.gz")
-    echo "{\"rev\": \"${commit_sha}\", \"sha256\": \"${digest}\", \"version\": \"${version_number}\"}" > $repo.json
+    echo "{\"rev\": \"${commit_sha}\", \"sha256\": \"${digest}\", \"version\": \"${version_number}\"}" > $repo-$branch.json
 }
 
-update_repo emacs-mirror emacs
+update_repo emacs-mirror emacs master
+update_repo emacs-mirror emacs emacs-27
```

Comment:
This emacs-27 branch is not a release, we'll also end up getting unreleased commits. If there was a good way to automatically find out the latest Emacs pre-release I'd be happy to add an `emacsUnstable` package.

Reply:
I understand the problem, you are right.
In the new version of my Github's fork you can see that now `emacsGit27` is not built from the HEAD of `emacs-27` branch but from a revision that I took from [here](https://github.com/emacs-mirror/emacs/releases).
The problem with this solution is that someone has to manually update `emacs-27.json` in this repository, if you think that this solution is acceptable let me know and I will rename the package as `emacsUnstable` and update the readme.

Reply:
I was thinking about adding emacs 27 a while ago, but I decided not to as there was no way to automatically maintain the derivation.
The intent of this overlay is to get as close to zero as possible in terms of maintenance time.

I'd be happy to accept a non-github source if that's what it takes.

Reply:
I understand what you mean, my solution would not be automatic.
Maybe I could edit the `update` script to grab for the releases list (still on github) and put in the json file the revision of the last release which name starts with `emacs-27` (at the moment there is only one).
This is not as clean as it is for the master branch but I can't think nothing better. Let me know if this solution is acceptable.
I'm a beginner with NixOS and I'm not an Emacs expert, I started using this overlay and I would really like to use Emacs 27. So forgive me for my inaccuracies.